### ISSUE TITLE
iPhone X: Drawer content should have additional padding in horizontal layouts.

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#12103]
## Steps to Reproduce

On the iPhone X, drawer content tiles should be inset with additional left/right padding (for LTR/RTL directionalities respectively) to account for the sensor housing notch.

See Apple's [Human Interface Guidelines for iPhone X](https://developer.apple.com/ios/human-interface-guidelines/overview/iphone-x).

Dependent on issue #12098. Get at this via `MediaQuery.of().padding`.

Example:

![screen shot 2017-09-13 at 13 18 44](https://user-images.githubusercontent.com/351029/30455738-e1ebc012-9955-11e7-9e3e-911dcdb2fc76.png)

## Flutter Doctor

```
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-AU, channel master)
    • Flutter at /Users/cbracken/src/flutter/flutter
    • Framework revision dd7e313317 (51 minutes ago), 2017-09-14 12:28:21 -0700
    • Engine revision 57a1445a45
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at /Users/cbracken/Library/Android/sdk
    • Platform android-25, build-tools 25.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.0)
    • Xcode at /Applications/Xcode9.0.app/Contents/Developer
    • Xcode 9.0, Build version 9A235
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1
```

**Comment by cbracken** [comment]
Drawer items are padded as of #13545. Work is still required for the heading area.

**Comment by jonahgreenthal** [comment]
I think there's either still a remaining issue with this or (more likely) I don't know what I'm doing. Here's a screenshot of my app:
![image](https://user-images.githubusercontent.com/737172/45925354-0bcd8880-bed9-11e8-89cc-99934d9168b6.png)
The white area above the blue should also be blue.


The structure is
* Drawer
  * ListView
     * Container with topCenter alignment and BoxDecoration for the dark blue background
       * DrawerHeader
         * CachedNetworkImage
     * ListTile
     * ListTile
     * …

**Comment by zoechi** [comment]
@jonahgreenthal Please create a new issue. This one is long closed.
